Sources/Calendar.php

```diff
@@ -365,6 +423,55 @@ function CalendarPost()
 	$context['linktree'][] = array(
 		'name' => $context['page_title'],
 	);
+
+	loadCSSFile('jquery-ui.datepicker.css', array('defer' => false), 'smf_datepicker');
+	loadCSSFile('jquery.timepicker.css', array('defer' => false), 'smf_timepicker');
+	loadJavascriptFile('jquery-ui.datepicker.min.js', array('defer' => true), 'smf_datepicker');
+	loadJavascriptFile('jquery.timepicker.min.js', array('defer' => true), 'smf_timepicker');
+	loadJavascriptFile('datepair.min.js', array('defer' => true), 'smf_datepair');
+	addInlineJavascript('
+	$("#allday").click(function(){   
+		$("#start_time").attr("disabled", this.checked);
+		$("#end_time").attr("disabled", this.checked);
+		$("#tz").attr("disabled", this.checked);
+	});
+	$( "#event_time_input .date_input" ).datepicker({
+		dateFormat: "yy-mm-dd",
```

Comment:
For whatever reason, the date picker script uses `yy` to indicate a four digit year number. As for the decision to use ISO format for dates, it was in part driven by JavaScript's relatively weak date parsing capabilities compared to PHP's. If it were possible to use whatever the user had chosen as their preferred date format, I would have just told the date picker script to expect input in that format. But there are plenty of possible combinations that the JavaScript can't handle, so I had to choose a single, reliable standard for that. ISO format is compact, unambiguous, and international, whereas `mm-dd-yyyy` format is idiosyncratically American and ambiguous when dealing with real values. An American would read `10-02-2016` as the second day of October in the year 2016, while a Brit would read it as the tenth day of February.